### ISSUE TITLE
Consider a database instead of json

### ISSUE THREAD

**Issue by NicEastvillage** [RLBot/RLBotCastingNotes#19]
The json file is growing quickly, but json is not a great long term. We should consider moving to a proper database at some point. This ofc comes with multiple complications - like where should it be hosted. For now I am just opening this issue for potential discussion.

**Comment by VirxEC** [comment]
Use postgresql & host it on Heroku - that should be just fine

**Comment by oxrock** [comment]
I get the impression that it would take a long time for it to grow to the point of being unwieldy. But future proofing is generally a good idea. Might be a nice easy project to tackle for a fledgling developer.

**Comment by kipje13** [comment]
While I agree that a relational database would be better suited for this data, a relational database can significantly improve the consistancy of the data. However, I am not sure if it would be worth the effort because it does make the entire setup much more complicated.

If anything we could write some automated tests to quickly validate the json when reviewing PRs.

**Comment by tarehart** [comment]
Yeah, and the current setup has some great properties:

- Free
- Review / approval workflow
- Loads instantly
- Fewer moving parts to break
- Easy to change data format
  - Don't need to alter an add / edit form
  - Don't need to write and test a SQL update script
- Don't need to manage / share database passwords
- Simple enough for anyone to take over / contribute to
- Changes fully auditable and synced with the UI code

**Comment by NicEastvillage** [comment]
Definitely. Those properties should not be taken for granted. It's not like I think we need to switch any time soon.
